pandaserver/dataservice/setupper_atlas_plugin.py

```diff
@@ -0,0 +1,2226 @@
+                                        changed = True
+                                # use default input endpoint
+                                if not changed:
+                                    ddm_id = tmp_site_spec.ddm_endpoints_input[scope_tmp_site_input].getDefaultRead()
+                                    self.logger.debug(f"use default_read {ddm_id} for tape prestaging")
+                            self.logger.debug(f"use {ddm_id} for tape prestaging")
+                            # register dataset locations
+                            is_ok = True
+                        else:
+                            is_ok = True
+                            # set sources to handle T2s in another cloud and to transfer dis datasets being split in multiple sites
+                            if not missing_at_t1:
+                                for tmp_ddm_id in ddm_id_list:
+                                    opt_source[tmp_ddm_id] = {"policy": 0}
+                            # T1 used as T2
+                            if (
+                                job.getCloud() != self.site_mapper.getSite(tmp_dst_id).cloud
+                                and (not dst_ddm_id.endswith("PRODDISK"))
+                                and (job.prodSourceLabel not in ["user", "panda"])
+                                and self.site_mapper.getSite(tmp_dst_id).cloud in ["US"]
+                            ):
+                                tmp_dst_site_spec = self.site_mapper.getSite(tmp_dst_id)
+                                scope_input, _ = select_scope(tmp_dst_site_spec, job.prodSourceLabel, job.job_label)
+                                se_tokens = tmp_dst_site_spec.setokens_input[scope_input]
+                                # use T1_PRODDISK
+                                if "ATLASPRODDISK" in se_tokens:
+                                    ddm_id = se_tokens["ATLASPRODDISK"]
+                            elif job.prodSourceLabel in ["user", "panda"]:
+                                # use DATADISK
+                                tmp_site_spec = self.site_mapper.getSite(job.computingSite)
+                                (
+                                    scope_tmp_site_input,
+                                    scope_tmp_site_output,
+                                ) = select_scope(tmp_site_spec, job.prodSourceLabel, job.job_label)
+                                if "ATLASDATADISK" in tmp_site_spec.setokens_input[scope_tmp_site_input]:
+                                    tmp_ddm_id = tmp_site_spec.setokens_input[scope_tmp_site_input]["ATLASDATADISK"]
+                                    if tmp_ddm_id in tmp_site_spec.ddm_endpoints_input[scope_tmp_site_input].getLocalEndPoints():
+                                        self.logger.debug(f"use {tmp_ddm_id} instead of {ddm_id} for analysis input staging")
+                                        ddm_id = tmp_ddm_id
+                        # set share and activity
```

Comment:
some prodsourcelabel to activity mapping function would be useful, instead of finding it here in the middle of a big function